### DISCUSSION TITLE
How do i get a handle to the Mainwindow datacontext in Mainactivity.cs

### DISCUSSION THREAD

**Discussion by SaverinOnRails** [AvaloniaUI/Avalonia#11739] [General]
I am trying to get location of the android app. How can i connect to the main data context from here. Or is there any other suitable way of accessing location?:
```
public class MainActivity : AvaloniaMainActivity
    {
        protected override void OnCreate(Bundle savedInstanceState)
        {
            base.OnCreate(savedInstanceState);  
          
        }
    }
```

**Comment by timunie** [comment]
I'd create an Interface or Service to register such things in the main project, see also [#androidnative](https://github.com/AvaloniaUI/Avalonia/discussions/11730#discussioncomment-6148829) for detailed explanation
